discovery_engine_core/core/src/document.rs

```diff
@@ -196,6 +197,20 @@ impl NewsResource {
     }
 }
 
+/// Deserialization of `DateTime<Utc>` with fallback.
+fn deserialize_date_time_with_fallback<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // const FORMAT: &str = "%Y-%m-%d %H:%M:%S";
+    const FORMAT: &str = "%Y-%m-%dT%H:%M:%S";
```

Comment:
this is the format which works for `NaiveDateTime` serialized by bincode, and then later deserialized into `String`